Ss 18 implement purchase repos

DanyloPlotskykh opened the pull request:
### Changes
added implementation of purchase order & purchase order record repositories

### Testing
implement purchase order and purchase order record repositories
…, tests is working correctly
![image](https://github.com/user-attachments/assets/e6d67113-83e6-4761-b2c9-a07fd521c2e4)

Closes [SS-18](https://sageteam.atlassian.net/browse/SS-18)

DanyloPlotskykh committed:
non working getbyfield recordrepo

DanyloPlotskykh committed:
SS_18 implement purchase order and purchase order record repositories, tests is working correctly

Sage-Cat commented on _server/src/Database/PurchaseOrdersRepository.hpp:
```diff
@@ -0,0 +1,29 @@
+#include "IDatabaseManager.hpp"
+#include "common/Entities/PurchaseOrder.hpp"
+
+using Purchaseorder  = Common::Entities::PurchaseOrder;
```
`PurchaseOrder`

DanyloPlotskykh replied:
done

Sage-Cat commented on conanfile.py:
```diff
@@ -23,8 +23,8 @@ class SageStorePkg(ConanFile):
             self.requires("boost/1.82.0")
-        if not self.is_qt_installed_manually():
-            self.requires("qt/6.6.2")
+        # if not self.is_qt_installed_manually():
```
Remove

DanyloPlotskykh replied:
done

Sage-Cat commented on _server/tests/unit/PurchaseOrdersRecordRepositoryTest.cpp:
```diff
@@ -0,0 +1,93 @@
+class PurchaseOrderRecordRepoTest : public ::testing::Test 
+{
+protected:
+    std::unique_ptr<PurchaseOrderRecordRepository> purchOrdRecRepo;
```
Do not make names short. Keep it `purchaseOrderRecordRepository`

DanyloPlotskykh replied:
done

Sage-Cat commented on _server/src/Database/PurchaseOrdersRepository.cpp:
Make some screenshots/server logs to demonstrate that ur code really works
Link this screenshots here in pull request and in Jira ticket

DanyloPlotskykh replied:
![image](https://github.com/user-attachments/assets/408dd76c-810c-4d42-9744-7358d1500784)

DanyloPlotskykh replied:
in the pull request, i have demonstrated tests screenshots already, maybe)

Sage-Cat commented on _server/src/Database/PurchaseOrdersRecordRepository.hpp:
```diff
@@ -0,0 +1,29 @@
+#include "IDatabaseManager.hpp"
+#include "common/Entities/PurchaseOrderRecord.hpp"
+
+using Purchaseorderrecord = Common::Entities::PurchaseOrderRecord;
```
`PurchaseOrderRecord`

DanyloPlotskykh replied:
done

Sage-Cat commented on _server/src/Database/PurchaseOrdersRecordRepository.hpp:
```diff
@@ -0,0 +1,29 @@
+    std::vector<Purchaseorderrecord> getAll() const override;
+
+private:
+    Purchaseorderrecord PurchFromCurrentRow(const std::shared_ptr<IQueryResult> &queryResult) const;
```
`PurchaseFromCurrentRow`

DanyloPlotskykh replied:
done

DanyloPlotskykh committed:
Implement base functionality for PurchaseOrdersRepository

implement classes with base query for PurchaseOrdersRepository and PurchaseOrdersRecordRepository
add unit test for it

Sage-Cat committed:
docs: update general info for project

Making README and dev docs up-to-date

Signed-off-by: Sage-Cat <sagecatbox@gmail.com>
